src/modules/roc_fec/target_openfec/roc_fec/of_decoder.cpp

```diff
@@ -71,6 +71,20 @@ bool OFDecoder::valid() const {
     return valid_;
 }
 
+size_t OFDecoder::max_source_block() const {
+    roc_panic_if_not(valid());
+
+    if (codec_id_ == OF_CODEC_REED_SOLOMON_GF_2_M_STABLE) {
+        return OF_REED_SOLOMON_MAX_NB_SOURCE_SYMBOLS_DEFAULT;
```

Comment:
I've checked OpenFEC sources and it appears that these defines are just defaults (as the name suggests).

OpenFEC provides two function to get and set these and other codec parameters:

* of_get_control_parameter
* of_set_control_parameter

However, of_set_control_parameter() is not implemented for LDPC and RS currently.

I suggest to do the following:

* use of_get_control_parameter() instead of defines;

* create an issue (in Roc) for:
  * crating a patch for OpenFEC that implements of_set_control_parameter()
  * using of_set_control_parameter() in Roc sender with the block sizes provided by user (to remove any limitation)
  * using of_set_control_parameter() in Roc receiver either with the block sizes provided by session (to remove any limitation) OR with the maximum block sizes set by user via the receiver config (to keep some limit possible but make it configurable)

Reply:
Discussed that we still need to validate that block size fits into FECFRAME packet limitations (1 or 2 bytes for ESI) so we still need a check. Leaving it as is for now, we'll add of_set_control_parameter() support later.

Reply:
#193